### PR TITLE
Allow direct copy of image or video from capture to WP media library.

### LINKED ISSUE #5560: Editor: Media picker silently fails to capture photo/video if app doesn't have Photos permissions
### Expected behavior

If the option to capture a photo/video is available when I add media to a post, I expect it to work or give a meaningful error message.
### Actual behavior

If I don't give the app permission to access Photos on my device, when I add media to a post I have access to my site's media library (as of #5338) but I am also given the option to capture a photo/video in the media picker. However, after I capture that photo/video nothing happens (the photo/video isn't added to my media library or to the post).

(If the app doesn't have permission to access the device Camera, a meaningful error message is shown.)
### Steps to reproduce the behavior
1. Open iOS Settings > Privacy and disable permission to the WordPress app for Photos. (Leave permission for Camera enabled.)
2. Open the WordPress app and start a new post.
3. Open the media picker to add media.
4. Tap the camera icon at the top of the media picker.
5. Take a photo or video and choose to use it.

Result: You are returned to the media picker and the photo/video is not added to the site's media library or the post.

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #5560

To test:
- Go to the settings app and block the access to Photo Library for the WordPress app
 - Start a new post or edit an old one
 - Tap on add media
 - Tap on the media capture on the top
 - Check if the new media is added to the WordPress media library
 - Tap on Done
 - See if media is correctly added
 - Go to App setting and change media related settings (maximum image size and geo-location) and check if all works correctly.

Needs review: @frosty

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m [hunk 1]
```diff
@@ -151,23 +151,40 @@ - (void)addImage:(UIImage *)image
...
+    } else {
+        NSString *fileName = [NSString stringWithFormat:@"%@_%@", [[NSProcessInfo processInfo] globallyUniqueString], @".jpg"];        
         NSURL *fileURL = [NSURL fileURLWithPath:[NSTemporaryDirectory() stringByAppendingPathComponent:fileName]];
         NSError *error;
         if ([image writeToURL:fileURL type:(__bridge NSString *)kUTTypeJPEG compressionQuality:0.9 metadata:metadata error:&error]){
-            return [PHAssetChangeRequest creationRequestForAssetFromImageAtFileURL:fileURL];
+            [self addMediaFromURL:fileURL completionBlock:completionBlock];
+        } else {
+            if (completionBlock) {
+                completionBlock(nil, error);;
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Media/MediaLibraryPickerDataSource.m]: Nitpick, two semicolons

File: WordPress/Classes/Extensions/NSURL+Exporters.swift [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+            exportVideoToURL(url,
+                targetUTI: targetUTI,
+                maximumResolution: maximumResolution,
+                stripGeoLocation: stripGeoLocation,
+                successHandler: successHandler,
+                errorHandler: errorHandler)
+        default:
+            errorHandler(error: errorForCode(.UnsupportedAssetType,
+                failureReason: NSLocalizedString("This media type is not supported on WordPress.",
+                                                 comment: "Error reason to display when exporting an unknow asset type from the device library")))
```
Comment (frosty) [inline WordPress/Classes/Extensions/NSURL+Exporters.swift]: "unknown"

File: WordPress/Classes/Extensions/NSURL+Exporters.swift [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+                    else {
+                        return CGSizeZero
+                }
+
+                return CGSize(width: pixelWidth, height: pixelHeight)
+            } else {
+                return CGSizeZero
+            }
+        }
+    }
```
Comment (frosty) [inline WordPress/Classes/Extensions/NSURL+Exporters.swift]: This getter (for `pixelSize` is a little hard to follow the flow because of all of the returns – you could perhaps restructure it a little so that there are only three:

```
var pixelSize: CGSize {
    get {
        if isVideo {
            let asset = AVAsset(URL: self)
            if let track = asset.tracksWithMediaType(AVMediaTypeVideo).first {
                return CGSizeApplyAffineTransform(track.naturalSize, track.preferredTransform)
            }
        } else if isImage {
            let options: [NSString:NSObject] = [kCGImageSourceShouldCache: false]
            if let imageSource = CGImageSourceCreateWithURL(self, nil),
                let imageDictionary = CGImageSourceCopyPropertiesAtIndex(imageSource, 0, options) as? NSDictionary,
                let pixelWidth = imageDictionary[kCGImagePropertyPixelWidth as NSString] as? Int,
                let pixelHeight = imageDictionary[kCGImagePropertyPixelHeight as NSString] as? Int {
                return CGSize(width: pixelWidth, height: pixelHeight)
            }
        }

        return CGSizeZero
    }
}
```

File: WordPress/Classes/Extensions/NSURL+Exporters.swift [hunk 4]
```diff
@@ -0,0 +1,325 @@
+import Foundation
+import Photos
+import MobileCoreServices
+import AVFoundation
+
+extension NSURL: ExportableAsset {
+
+    func exportToURL(url: NSURL,
```
Comment (frosty) [inline WordPress/Classes/Extensions/NSURL+Exporters.swift]: It might be useful to add some documentation to some of these methods.

Comment (SergioEstevao) [inline WordPress/Classes/Extensions/NSURL+Exporters.swift]: The protocol has the documentation for the methods.

**Comment by SergioEstevao** [comment]
@frosty ready for another look.

**Comment by frosty** [comment]
Changes look good, @SergioEstevao! Definitely works better with the async thumbnail generation.

:shipit:
